src/db.rs

```diff
@@ -95,7 +95,13 @@ impl DbInner {
         key: &[u8],
         options: &ReadOptions,
     ) -> Result<Option<Bytes>, SlateDBError> {
-        let snapshot = self.state.read().snapshot();
+        let snapshot = {
```

Comment:
`read` is a method provided by `RwLock`, not `DbState`, so it's not as simple. I guess the question I'd have is whether we care about acquiring the lock before checking the error or not. I assumed we want to acquire the lock before, but if that's not the case, this could be implemented like this:

```rust
if let Some(err) = self.state.error() {
  return Err(...)
}
let snapshot = self.state.read().snapshot();
```